### PR TITLE
[feat]: [IAC-50]: Add unittest for the IACM manager

### PR DESCRIPTION
**Pull request by nassergonzalez**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)

### CODE REVIEW

**Comment by nassergonzalez** [comment]
trigger ft1

**Comment by nassergonzalez** [comment]
trigger ft2

File: 310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            return false;
+          }
+        });
+      }
+    });
+
+    IACMManagerConfiguration configuration =
+        IACMManagerConfiguration.builder()
+            .managerAuthority("localhost")
+            .managerTarget("localhost:9880")
```
Comment (allofthesepeople) [inline 310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java:156]: assume these are hard coded in the wider test suite? Or could we have them as vars?

**Review by allofthesepeople — APPROVED** [review]
one question on setting the tests urls as a vars rather than hard-coded but guessing it's I-keeping with the rest of the test harness

File: 310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java (hunk 1 above)
Comment (nassergonzalez) [reply 310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java:156]: The first one, they are hard coded in the other tests too

**Review by willd0yle — DISMISSED** [review]

**Review by nassergonzalez — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192#-:-NIasACc3yIFNU_zKpTj:bnfp4nl)* status: 16 file(s) to review. 1. unresolved discussions.

**Review by nassergonzalez — COMMENTED** [review]
> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: 16 file(s) to review. 1. unresolved discussions.

**Review by naidu-harness — DISMISSED** [review]
> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by allofthesepeople — COMMENTED** [review]
> Reviewed 16 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: Some of the checks are too old. 1. unresolved discussions.

**Review by allofthesepeople — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: Some of the checks are too old. 

___
*[`310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java` line 156 at r1](https://reviewable.io/reviews/harness/harness-core/40192#-NGvlnyEztBNLtdWudvO-r1-156:-NJ6I_gVD3TC73-F7Z5P:bb6ywmx) ([raw file](https://github.com/harness/harness-core/blob/8c2e299a1cc8608a3f89d72d3303935ea5c4c3f0/310-iacm-manager/src/test/java/io/harness/app/impl/IACMManagerRule.java#L156)):*
<details><summary><i>Previously, nassergonzalez (Nasser Gonzalez Gonzalez) wrote…</i></summary><blockquote>

The first one, they are hard coded in the other tests too
</blockquote></details>

LGTM

**Review by nassergonzalez — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192#-:-NJ6K2Mr6XHOeBiQE4GF:bnfp4nl)* status: Some of the checks are too old.

**Review by allofthesepeople — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192#-:-NJ6Kgh0335p-0VYm1iH:bnfp4nl)* status: Some of the checks are too old.

**Review by RickCraig — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: Some of the checks are too old.

**Review by RickCraig — DISMISSED** [review]
> Reviewed 16 of 16 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: Some of the checks are too old.

**Comment by nassergonzalez** [comment]
trigger smartchecks

**Comment by nassergonzalez** [comment]
trigger ti0 trigger ti1 trigger ti2 trigger ti3 trigger ti4 trigger ft1 trigger ft2

**Comment by nassergonzalez** [comment]
trigger checkstylecodeformat trigger pmd

**Comment by nassergonzalez** [comment]
trigger codebasehash

**Review by allofthesepeople — APPROVED** [review]

**Review by allofthesepeople — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40192)* status: :shipit: complete! ✓

**Pull request merged**